### PR TITLE
add warning from issue #2495

### PR DESCRIPTION
**Pull request by mnsn**
**Summary**

as request in issue #2495 we should warn user when installing yarn globally using yarn, I have added a warning massage in the right place
**Test plan**
no test required

### CODE REVIEW

File: src/reporters/lang/en.js [hunk 1]
```diff
@@ -238,17 +238,16 @@ const messages = {
...
-
+  
   unknownFetcherFor: 'Unknown fetcher for $0',
-
+  
   refusingDownloadGitWithoutCommit: 'Refusing to download the git repo $0 over plain git without a commit hash',
   refusingDownloadHTTPWithoutCommit: 'Refusing to download the git repo $0 over HTTP without a commit hash',
   refusingDownloadHTTPSWithoutCommit: 'Refusing to download the git repo $0 over HTTPS without a commit hash - possible certificate error?',
-
+
```
Comment (Daniel15) [inline src/reporters/lang/en.js]: Could you please revert all the unnecessary changes, such as these spaces? In general, the Facebook coding style (which is somewhat used in the Yarn codebase too) is to trim all trailing whitespace from every line, so blank lines like these should be totally blank.

File: src/reporters/lang/en.js [hunk 2]
```diff
@@ -257,6 +256,7 @@ const messages = {
   optionalDepNotInstalled: 'Optional dependency $0 not installed',
   packageWrongVersion: '$0 is wrong version: expected $1, got $2',
   packageDontSatisfy: '$0 doesn\'t satisfy found match of $1',
+  packageContainsYarnAsGlobal: 'installing Yarn via Yarn will result in you having two separate versions of Yarn installed at the same time.',
```
Comment (Daniel15) [inline src/reporters/lang/en.js]: Maybe also mention that it's not recommended, and to install Yarn by following the instructions at https://yarnpkg.com/en/docs/install?

Comment (mnsn) [inline src/reporters/lang/en.js]: OK so the full text will be:
installing Yarn via Yarn will result in you having two separate versions of Yarn installed at the same time, which is not recommended. To update Yarn please follow https://yarnpkg.com/en/docs/install

**Comment by mnsn** [comment]
Hi @Daniel15 thank you for your review I've added the spaces and changed the message.

File: src/cli/commands/global.js [hunk 3]
```diff
@@ -178,7 +178,11 @@ const {run, setFlags: _setFlags} = buildSubCommands('global', {
     await updateCwd(config);
 
     const updateBins = await initUpdateBins(config, reporter, flags);
-
+
```
Comment (bestander) [inline src/cli/commands/global.js]: nit: a few whitespaces got into this line

File: src/cli/commands/global.js [hunk 4]
```diff
@@ -178,7 +178,11 @@ const {run, setFlags: _setFlags} = buildSubCommands('global', {
     await updateCwd(config);
 
     const updateBins = await initUpdateBins(config, reporter, flags);
-
+    
+    if (args.includes('yarn')) {
+      reporter.warn(reporter.lang('packageContainsYarnAsGlobal'));
+
```
Comment (bestander) [inline src/cli/commands/global.js]: nit: this empty line not needed here

Comment (mnsn) [inline src/cli/commands/global.js]: Thank you @bestander , I removed the empty line

Comment (bestander) [inline src/cli/commands/global.js]: don't see it :)

Comment (mnsn) [inline src/cli/commands/global.js]: Sorry did it on my master now it's in

File: src/cli/commands/global.js [hunk 5]
```diff
@@ -176,9 +176,11 @@ const {run, setFlags: _setFlags} = buildSubCommands('global', {
     args: Array<string>,
   ): Promise<void> {
     await updateCwd(config);
-
     const updateBins = await initUpdateBins(config, reporter, flags);
-
+    if (args.includes('yarn')) {
+      reporter.warn(reporter.lang('packageContainsYarnAsGlobal'));
+
```
Comment (bestander) [inline src/cli/commands/global.js]: one more space

**Comment by bestander** [comment]
Just one more space is unnecessary and could you rebase please?
CI had some stability fixes recently

**Comment by bestander** [comment]
Fixed in #2926
